src/github-provider.ts

```diff
@@ -104,6 +131,13 @@ function GithubProvider(this: any, _options: any) {
     }
 
     octokit = new Octokit(config)
+
+    actions = {
```

Comment:
seems unnecessary? 

`octokit.rest[entityOctoName][octoOperation]` then resolve `entityOctoName` and `octoOperation` when building the action functions and capture them in closure

Reply:
totally agree ! thanks

Reply:
That are cases where `octokit.rest[entityOctoName][octoOperation]` won't work well. For example : you'll find branch related methods in  `octokit.rest.repos.getBranch` or commits related methods in `octokit.rest.git.getCommit`. There are also situations related to plural like codesOfConduct and codeScanning. 
I created a rest_endpoint to be filled in entities data structure. What do you think about? Sure, I can change it, anyway.

Reply:
Richard, `entityOctoName` and `octoOperation` would come from the data structures in this case ? I mean, manually ? I did it this way, but sure I'm open to change it anyway.
I tried to resolve `octokit.rest[entityOctoName][octoOperation]` only by the entity name but it did work well. For example : you'll find branch related methods in `octokit.rest.repos.getBranch` or commits related methods in `octokit.rest.git.getCommit.` There are also situations related to plural like codesOfConduct and codeScanning.